src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -6576,6 +6580,38 @@ static void Main()
             var verifier = CompileAndVerify(comp, expectedOutput: "3");
         }
 
+        [Fact]
+        public void DeconstructionWarnsForSelfAssignment()
```

Comment:
Yes, thanks for pointing that out. That will still be allowed, as it can be useful.